fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQueryRenderer.java

```diff
@@ -385,49 +385,61 @@ protected WhereFragment getTokenFilter(QueryParameter queryParm, String paramAli
                         where.col(paramAlias, COMMON_TOKEN_VALUE_ID).eq(nullCheck(commonTokenValueId));
                     }
                 } else {
-                    // traditional approach, using a join to xx_TOKEN_VALUES_V
-                    where.col(paramAlias, "TOKEN_VALUE").operator(operator);
-                    if (operator == Operator.LIKE) {
-                        // Must escape special wildcard characters _ and % in the parameter value string.
-                        String textSearchString = SqlParameterEncoder.encode(value.getValueCode())
-                                .replace(PERCENT_WILDCARD, ESCAPE_PERCENT)
-                                .replace(UNDERSCORE_WILDCARD, ESCAPE_UNDERSCORE)
-                                .replace("+", "++")+ PERCENT_WILDCARD;
-                        where.bind(SearchUtil.normalizeForSearch(textSearchString)).escape("+");
-
-                    } else {
-                        where.bind(SqlParameterEncoder.encode(value.getValueCode()));
+                    // Traditional approach, using a join to xx_TOKEN_VALUES_V
+                    
+                    // Include code if present
+                    if (code != null) {
+                        where.col(paramAlias, TOKEN_VALUE).operator(operator);
+                        if (operator == Operator.LIKE) {
+                            // Must escape special wildcard characters _ and % in the parameter value string.
+                            String textSearchString = normalizedCode
+                                    .replace(PERCENT_WILDCARD, ESCAPE_PERCENT)
+                                    .replace(UNDERSCORE_WILDCARD, ESCAPE_UNDERSCORE)
+                                    .replace("+", "++")+ PERCENT_WILDCARD;
+                            where.bind(SearchUtil.normalizeForSearch(textSearchString)).escape("+");
+    
+                        } else {
+                            where.bind(normalizedCode);
+                        }
                     }
 
-                    // Include system if present.
-                    if (value.getValueSystem() != null && !value.getValueSystem().isEmpty()) {
-                        if (operator == Operator.NE) {
-                            where.or();
-                        } else {
+                    // Include system if present
+                    if (system != null) {
+                        if (code != null) {
                             where.and();
                         }
 
-                        // filter on the code system for the given parameter
-                        Integer codeSystemId = identityCache.getCodeSystemId(value.getValueSystem());
+                        // Filter on the code system for the given parameter
+                        Integer codeSystemId = identityCache.getCodeSystemId(system);
                         where.col(paramAlias, CODE_SYSTEM_ID).operator(operator).literal(nullCheck(codeSystemId));
```

Comment:
The codeSystemId value is a normalized database identifier. I don't think it makes sense to use anything other an equals here. I would make this `...eq().literal(nullCheck(codeSystemId));`